pkg/aerospike/discovery.go

```diff
@@ -75,15 +76,47 @@ func (conf *AerospikeProbeConfig) generateNamespacedEndpointsFromEntry(logger lo
 	return endpoints, nil
 }
 
-func (conf AerospikeProbeConfig) getNamespacesFromEntry(entry discovery.ServiceEntry) map[string]struct{} {
+func (conf AerospikeProbeConfig) getNamespacesFromEntry(logger log.Logger, entry discovery.ServiceEntry) map[string]struct{} {
 	namespaces := make(map[string]struct{})
-	nsString, ok := entry.Meta[conf.AerospikeEndpointConfig.NamespaceMetaKey]
-	if ok {
-		nsFromDiscovery := strings.Split(nsString, ";")
-		for _, ns := range nsFromDiscovery {
-			namespaces[ns] = struct{}{}
+	fallback := false
+
+	// Correct way to get namespaces.
+	for metaKey, metaValue := range entry.Meta {
+		if !strings.HasPrefix(metaKey, conf.AerospikeEndpointConfig.NamespaceMetaKeyPrefix) {
+			continue
+		}
+		ready, err := strconv.ParseBool(metaValue)
+		// if the value of the NamespaceMetaKeyPrefix MetaData is not a boolean then fallback to the old method
+		if err != nil {
+			level.Warn(logger).Log("Fail to parse boolean value from MetaData %s. Will fallback to deprecated %s",
+				metaKey, conf.AerospikeEndpointConfig.NamespaceMetaKeyPrefix)
+			level.Warn(logger).Log(err.Error())
```

Comment:
Log format is not consistent.
should look like this:
level.Error(e.Logger).Log("msg", fmt.Sprintf("Error while fetching record: %s", keyAsStr(key)), "err", err)
With key=value pairs and a single line

Reply:
Done, thanks !